torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -297,15 +322,26 @@ void TensorPipeAgent::startImpl() {
       // a channel that comes before another receives a higher priority.
       priority = opts_.channels->size() - 1 - (iter - opts_.channels->begin());
     }
-    std::unique_ptr<ChannelRegistration> reg =
-        TensorPipeChannelRegistry()->Create(key);
+    std::unique_ptr<CpuChannelRegistration> reg =
+        TensorPipeCpuChannelRegistry()->Create(key);
     if (priority == -1) {
       priority = reg->priority;
     }
     context_->registerChannel(
         priority, std::move(key), std::move(reg->channel));
   }
 
+#if TENSORPIPE_USE_CUDA_IPC_CHANNEL
+
+  for (auto& key : TensorPipeCudaChannelRegistry()->Keys()) {
+    std::unique_ptr<CudaChannelRegistration> reg =
```

Comment:
The CPU version of this code has some logic to allow users to override the priorities of channels through a (hidden) field of the options struct. Should we add the same logic here?